File: tests/fixtures/typescript/fields.ts
import { AuditLog, UserRepository } from "./repos";

export class Base {
  audit = new AuditLog();
}

export class Issue extends Base {
  title: string;
  posterId = 0;
  labels: string[] = [];

  constructor(
    private repo: UserRepository,
    private log: AuditLog,
  ) {
    super();
    this.title = "";
    this.close = this.close.bind(this);
  }

  close(): void {
    this.labels = [];
    this.audit = new AuditLog();
    void this.repo.deleteUser(this.posterId);
    this.log.deleteUser(this.posterId);
  }

  label(kind: string): string {
    switch (kind) {
      case "poster": {
        return String(this.posterId);
      }
      case "copy": return {
        posterId: 1,
        label(): string {
          return `${this.posterId}`;
        },
      }.label();
      default: {
        return this.title;
      }
    }
  }
}

export interface Comment {
  posterId: number;
  body: string;
}

export function show(issue: Issue, comment: any): void {
  console.log(issue.posterId, issue.title, issue.audit);
  console.log(comment.posterId, comment.body);
}

export function tally(comments: any[]): number {
  let total = 0;
  const sums = {
    total: 0,
  };
  for (const comment of comments) {
    total += comment.total + sums.total;
  }
  return total;
}

File: tests/fixtures/python/fields.py
from repos import AuditLog, UserRepository


class Base:
    def __init__(self) -> None:
        self.audit = AuditLog()


class Issue(Base):
    title: str
    poster_id: int = 0

    def __init__(self, repo: UserRepository) -> None:
        super().__init__()
        self.repo = repo
        self.labels: list[str] = []

    async def close(self) -> None:
        if self.labels:
            self.labels = []
        await self.repo.delete_user(self.poster_id)


class Comment:
    def __init__(self, poster_id: int, body: str) -> None:
        self.poster_id = poster_id
        self.body = body


def show(issue: Issue, comment) -> None:
    print(issue.poster_id, issue.title, issue.audit)
    print(comment.poster_id, comment.body)


def tally(comments) -> int:
    total: int = 0
    for comment in comments:
        total += comment.total
    return total
